### PR TITLE
Suse manager settings refactor

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

Naming and structural refactor of SUMACredentials.

The concept of SUMACredentials is replaced by SuseManagerSettings, this choice is made to unify all the settings under the same context `settings` and under the same controller/policy `SettingsController`/`Settings.Policy`

- Suse manager settings moved under the settings context
- Removed Suse credentials controller
- Web actions of suse manager moved under settings controller
- OpenApi renaming 
- New endpoints created under `/settings/suse_manager`, `settings/suma_credentials` endpoint marked as deprecated, backward compatibility guaranteed
- Frontend consumes the new endpoints
- Renamed suma credentials to suse manager settings through the whole codebase, tests included
- Controller views unified

## How was this tested?

Automated tests

### CODE REVIEW

File: lib/trento_web/router.ex [hunk 1]
```diff
@@ -183,12 +183,23 @@ defmodule TrentoWeb.Router do
...
+            get "/", SettingsController, :get_suse_manager_settings
+            post "/", SettingsController, :save_suse_manager_settings
+            patch "/", SettingsController, :update_suse_manager_settings
+            put "/", SettingsController, :update_suse_manager_settings
+            delete "/", SettingsController, :delete_suse_manager_settings
+            post "/test", SettingsController, :test_suse_manager_settings
+          end
 
-            post "/test", SUMACredentialsController, :test
+          # deprecated
```
Comment (arbulu89) [inline lib/trento_web/router.ex:195]: It's a pity that we cannot really deprecate the whole endpoint in openapi itself.
This way, the path is not shown as deprecated.
I have looked a bit, and didn't find any function to do so, without deprecating the new one as well

**Review by arbulu89 — APPROVED** [review]
Great work!

**Review by dottorblaster — APPROVED** [review]
> Because he's the hero Gotham deserves, but not the one it needs right now. So we'll hunt him. Because he can take it. Because he's not our hero. He's a silent guardian, a watchful protector. A dark knight.

File: assets/js/lib/api/softwareUpdatesSettings.js [hunk 2]
```diff
@@ -1,16 +1,15 @@
 import { networkClient } from '@lib/network';
 
-export const getSettings = () =>
-  networkClient.get(`/settings/suma_credentials`);
+export const getSettings = () => networkClient.get(`/settings/suse_manager`);
```
[the remark this answers is not in the document]
Comment (CDimonaco) [reply assets/js/lib/api/softwareUpdatesSettings.js]: done, nice catch!

**Pull request merged**